packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/apploader/RNHeadlessAppStarter.kt

```diff
@@ -0,0 +1,55 @@
+package expo.modules.taskManager.apploader
+
+import android.content.Context
+import com.facebook.react.ReactApplication
+import com.facebook.react.ReactInstanceManager
+import expo.loaders.provider.interfaces.HeadlessAppStarter
+import expo.modules.taskManager.TaskService
+import org.unimodules.core.interfaces.Consumer
+import java.util.*
+
+private val appRecords: MutableMap<String, ReactInstanceManager> = HashMap()
+
+class RNHeadlessAppStarter(private val ignore: Context) : HeadlessAppStarter {
+
+  //region AppLoader
+
+  override fun startApp(context: Context, params: HeadlessAppStarter.Params?, alreadyRunning: Runnable?, callback: Consumer<Boolean>?) {
+    if (params == null || params.appId == null) {
+      throw IllegalArgumentException("Params must be set with appId!")
+    }
+
+    val reactInstanceManager = (context.applicationContext as ReactApplication).reactNativeHost.reactInstanceManager
```

Comment:
Are we pretty much guaranteed than in any case (even anyone's customized bare RN app) the context we receive can be cast to `ReactApplication`? I tried quickly to trace back where this context comes from but it's not obvious. 🤔  If not, it might be nice to throw a helpful error here and also be sure to document this assumption.